cmd/shipperctl/cmd/backup/restore.go

```diff
@@ -0,0 +1,201 @@
+package backup
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/yaml"
+
+	"github.com/bookingcom/shipper/cmd/shipperctl/configurator"
+	"github.com/bookingcom/shipper/cmd/shipperctl/ui"
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+	"github.com/bookingcom/shipper/pkg/client/clientset/versioned"
+)
+
+var (
+	backupFile string
+
+	restoreBackupCmd = &cobra.Command{
+		Use:   "restore",
+		Short: "Restore backup that was prepared using `shipperctl prepare` command",
+		Long: "Restore backup that was prepared using `shipperctl prepare` command" +
+			"this will update the owner reference of releases objects",
+		RunE: runRestoreCommand,
+	}
+)
+
+func init() {
+	BackupCmd.AddCommand(restoreBackupCmd)
+}
+
+func runRestoreCommand(cmd *cobra.Command, args []string) error {
+	kubeClient, err := configurator.NewKubeClientFromKubeConfig(kubeConfigFile, managementClusterContext)
+	if err != nil {
+		return err
+	}
+
+	shipperClient, err := configurator.NewShipperClientFromKubeConfig(kubeConfigFile, managementClusterContext)
+	if err != nil {
+		return err
+	}
+
+	releasesPerApplications, err := unmarshalReleasesPerApplicationFromFile()
+	if err != nil {
+		return err
+	}
+
+	confirm, err := ui.AskForConfirmation(cmd.InOrStdin(), "Would you like to see an overview of your backup?")
+	if err != nil {
+		return err
+	}
+	if confirm {
+		printReleasesPerApplications(releasesPerApplications)
+		confirm, err = ui.AskForConfirmation(cmd.InOrStdin(), "Would you like to review backup?")
+		if err != nil {
+			return err
+		}
+		if confirm {
+			data, err := marshalReleasesPerApplications(releasesPerApplications)
```

Comment:
What's the reason we asked again to print the releases here?

Reply:
The first is an overview: a table of the releases and their application
The second is the backup file itself

Reply:
👍